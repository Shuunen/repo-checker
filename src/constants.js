import { join } from 'path'

export const home = process.env.HOME
export const dataFileName = 'repo-checker.config.js'
export const repoCheckerPath = join(__dirname, '..')
export const templatePath = join(repoCheckerPath, 'templates')
export const dataFileHomePath = join(home, dataFileName)

export const dataDefaults = {
  auto_merge: true,
  ban_sass: true,
  dev_deps_only: true,
  license: 'GPL-3.0',
  max_size_ko: 10,
  user_mail: 'romain.racamier@gmail.com',
  user_name: 'Romain Racamier-Lafon',
}
