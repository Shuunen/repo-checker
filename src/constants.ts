import { join } from 'path'

/* istanbul ignore next */
export const home = process.env.HOME ?? '' // when does HOME is not defined ?
export const dataFileName = 'repo-checker.config.js'
/* istanbul ignore next */
export const repoCheckerPath = process.env.pwd ?? process.cwd()
export const templatePath = join(repoCheckerPath, 'templates')
export const dataFileHomePath = join(home, dataFileName)

export class ProjectData {
  auto_merge = true
  ban_sass = true
  dev_deps_only = true
  is_module = false
  license = 'GPL-3.0'
  max_size_ko = 1
  npm_package = false
  package_name = ''
  quiet = false
  repo_id = 'a-great-repo'
  use_typescript = false
  use_vue = false
  user_id = 'Shuunen'
  user_id_lowercase = 'shuunen'
  user_mail = 'romain.racamier@gmail.com'
  user_name = 'Romain Racamier-Lafon'
  web_published = false
  web_url = 'https://my-website.com'

  constructor (data: Partial<ProjectData> = {}) {
    Object.assign(this, data)
  }
}

export const dataDefaults = new ProjectData()
