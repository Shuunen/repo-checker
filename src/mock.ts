/* eslint-disable jsdoc/require-jsdoc */
import path from 'node:path'
import { clone, sleep } from 'shuutils'
import type { Indicators } from './check.ts'
import type { FileBase } from './file.ts'
import { join } from './utils.ts'

const fileBaseKeysToDelete: (keyof FileBase)[] = ['fileExists', 'checkFileExists', 'folderPath', 'inspectFile', 'updateFile', 'initFile']
const indicatorsKeys: ReadonlyArray<keyof Indicators> = ['failed', 'passed', 'warnings'] as const

function cleanStringForSnap(input: string) {
  return (
    input
      // replace repo-check-1-40 by repo-check-x-yy
      .replace(/repo-check-\d-\d+/gu, 'repo-check-x-yy')
      // replace package-json-main-file-size-63ko by package-json-main-file-size-XYZko
      .replace(/package-json-main-file-size-\d+ko/gu, 'package-json-main-file-size-XYZko')
  )
}

// eslint-disable-next-line no-restricted-syntax
export function cleanUnknownValueForSnap<Type>(input: Type): Type {
  /* c8 ignore next 8 */
  /* eslint-disable @typescript-eslint/consistent-type-assertions, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-type-assertion */
  if (typeof input === 'string') return cleanStringForSnap(input) as Type
  if (Array.isArray(input)) return input.map(item => cleanUnknownValueForSnap(item)) as Type
  if (typeof input === 'object') return JSON.parse(cleanStringForSnap(JSON.stringify(input))) as Type
  // eslint-disable-next-line no-restricted-syntax
  throw new Error(`cleanInstanceValueForSnap: unknown type ${typeof input}`)
  /* eslint-enable @typescript-eslint/consistent-type-assertions, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-type-assertion */
}

export async function promiseValue<Type>(value: Type) {
  await sleep(1)
  return value
}

export async function promiseTrue() {
  await sleep(1)
  return true
}

export async function promiseFalse() {
  await sleep(1)
  return false
}

export async function promiseVoid() {
  await sleep(1)
}

// eslint-disable-next-line unicorn/prefer-module
export const sourceFolder = path.dirname(__filename)

export const vueProjectFolder = join(sourceFolder, 'mocks', 'vueProject')

export const tsProjectFolder = join(sourceFolder, 'mocks', 'tsProject')

export const mocksProjectsFolder = join(sourceFolder, 'mocks')

export function cleanIndicatorsForSnap(record?: Readonly<Record<string, unknown>>) {
  if (!record) return record
  const clean = clone(record)
  for (const key of indicatorsKeys) if (key in clean) clean[key] = cleanUnknownValueForSnap(clean[key])
  return clean
}

// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
export function cleanInstanceForSnap(instance: Readonly<FileBase>, ...bonusKeys: readonly string[]) {
  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
  const clean = clone<Partial<FileBase>>(instance) as Record<string, unknown>
  const keys = [...fileBaseKeysToDelete, ...bonusKeys] // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
  for (const key of keys) if (key in clean) delete clean[key]
  return cleanIndicatorsForSnap(clean)
}
