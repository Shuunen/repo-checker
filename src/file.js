import path from 'path'
import { templatePath } from './constants'
import { log } from './logger'
import { createFile, fillTemplate, folderContainsFile, getFileSizeInKo, readFileInFolder } from './utils'

export class File {
  constructor (folderPath, data, doFix, doForce) {
    this.folderPath = folderPath
    this.data = data
    this.doFix = doFix
    this.doForce = doForce
    this.fileContent = ''
    this.originalFileContent = ''
    this.fileName = ''
    this.nbPassed = 0
    this.nbFailed = 0
  }

  async end () {
    await this.updateFile()
    await this.checkIssues()
  }

  async inspectFile (fileName) {
    this.fileName = fileName
    this.originalFileContent = this.fileContent = await readFileInFolder(this.folderPath, fileName, true)
  }

  async updateFile () {
    if (!this.doFix) return
    if (this.originalFileContent === this.fileContent) return
    return createFile(this.folderPath, this.fileName, this.fileContent)
  }

  async checkFileExists (fileName, justWarn = false) {
    let fileExists = await folderContainsFile(this.folderPath, fileName)
    if (!fileExists && this.doFix) {
      const fileContent = await this.createFile(fileName)
      fileExists = fileContent.length > 0
    }
    this.test(fileExists, `has a ${fileName} file`, justWarn)
    return !!fileExists
  }

  async checkNoFileExists (fileName, justWarn = false) {
    const fileExists = await folderContainsFile(this.folderPath, fileName)
    this.test(!fileExists, `has no ${fileName} file`, justWarn)
  }

  async createFile (fileName) {
    const template = await readFileInFolder(templatePath, fileName, true)
    const fileContent = fillTemplate(template, this.data)
    if (fileContent.length > 0) {
      await createFile(this.folderPath, fileName, fileContent)
      log.fix('created', fileName)
    } else {
      log.warn(`please provide a data file to be able to fix a "${fileName}" file`)
    }
    return fileContent
  }

  async checkIssues () {
    if (this.nbFailed > 0 && this.doFix) {
      if (this.doForce) return this.createFile(this.fileName)
      log.info('this file has at least one issue, if you want repo-checker to overwrite this file use --force')
    }
  }

  async getFileSizeInKo (filePath) {
    return getFileSizeInKo(path.join(this.folderPath, filePath))
  }

  /**
   * shouldContains verify that the target file contains specific strings
   * @param {string} name The name of the test
   * @param {regex} regex The regex that should validate the test
   * @param {number} nbMatchExpected The number of matches expected
   * @param {boolean} justWarn If this test is optional
   * @return a boolean which indicate if the content exists
   */
  shouldContains (name, regex, nbMatchExpected = 1, justWarn = false, helpMessage = '', canFix = false) {
    if (!regex) regex = new RegExp(name)
    const contentExists = this.checkContains(regex, nbMatchExpected)
    const fix = this.doFix && canFix && !contentExists
    name += (contentExists || fix) ? '' : ` -- ${helpMessage || regex.toString().replace(/\\/g, '')}`
    const message = `${this.fileName} ${!contentExists ? (justWarn ? 'could have' : 'does not have') : 'has'} ${name} `
    if (fix) {
      log.fix(message)
    } else {
      this.test(contentExists, message, justWarn)
    }
    return contentExists
  }

  couldContains (name, regex, nbMatchExpected = 1, helpMessage = '', canFix = false) {
    return this.shouldContains(name, regex, nbMatchExpected, true, helpMessage, canFix)
  }

  checkContains (regex, nbMatchExpected) {
    const matches = this.fileContent.match(regex)
    const nbMatch = (matches && matches.length) || 0
    if (nbMatchExpected && nbMatch !== nbMatchExpected) {
      log.debug(regex.toString().replace('\n', ''), 'matched', nbMatch, 'instead of', nbMatchExpected)
    }
    return (nbMatch === nbMatchExpected)
  }

  test (isValid, message, justWarn = false) {
    if (!isValid && !justWarn) {
      this.nbFailed++
    } else {
      this.nbPassed++
    }
    log.test(isValid, message, justWarn)
  }
}
