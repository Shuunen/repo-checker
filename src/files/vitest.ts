import { FileBase } from '../file.ts'

// eslint-disable-next-line no-export
export class VitestFile extends FileBase {
  // eslint-disable-next-line jsdoc/require-jsdoc
  // eslint-disable-next-line max-lines-per-function
  public async start() {
    if (!this.data.isUsingVitest) return
    const hasMtsFile = await this.fileExists('vitest.config.mts')
    const hasJsFile = await this.fileExists('vitest.config.js')
    const hasTsFile = await this.fileExists('vitest.config.ts')
    const hasMjsFile = await this.fileExists('vitest.config.mjs')
    if (hasMtsFile) await this.inspectFile('vitest.config.mts')
    /* c8 ignore next 3 */
    if (hasJsFile) await this.inspectFile('vitest.config.js')
    if (hasTsFile) await this.inspectFile('vitest.config.ts')
    if (hasMjsFile) await this.inspectFile('vitest.config.mjs')
    this.couldContains('a pool thread setting', /pool: 'threads'/u)
  }
}
