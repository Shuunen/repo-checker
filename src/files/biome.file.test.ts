import { expect, it } from 'vitest'
import { ProjectData, repoCheckerPath } from '../constants'
import { cleanInstanceForSnap, promiseFalse, promiseTrue, promiseVoid } from '../mock'
import { BiomeFile } from './biome.file'

it('biome A not using biome', async () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  instance.data.isUsingBiome = false
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('biome B missing file', async () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  instance.data.isUsingBiome = true
  instance.checkFileExists = promiseFalse
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('biome C empty file', async () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  instance.data.isUsingBiome = true
  instance.checkFileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = ''
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('biome D invalid json', async () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  instance.data.isUsingBiome = true
  instance.checkFileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = '{ invalid json }'
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('biome E minimal valid biome.json', async () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  instance.data.isUsingBiome = true
  instance.checkFileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = '{"$schema":"./node_modules/@biomejs/biome/configuration_schema.json"}'
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('biome F fix missing sections', async () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }), true)
  instance.data.isUsingBiome = true
  instance.checkFileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = '{"$schema":"./node_modules/@biomejs/biome/configuration_schema.json"}'
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('biome G full valid biome.json', async () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  instance.data.isUsingBiome = true
  instance.checkFileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = JSON.stringify({
    $schema: './node_modules/@biomejs/biome/configuration_schema.json',
    formatter: { indentStyle: 'space', indentWidth: 2, lineEnding: 'lf', lineWidth: 120 },
    javascript: { formatter: { arrowParentheses: 'asNeeded', bracketSpacing: true, quoteStyle: 'single', semicolons: 'asNeeded' } },
    json: { parser: { allowComments: false, allowTrailingCommas: false } },
    linter: { enabled: true, rules: { recommended: true, style: { useBlockStatements: 'off' } } },
    organizeImports: { enabled: true },
    vcs: { clientKind: 'git', enabled: true, useIgnoreFile: true },
  })
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('biome H throws if javascript section is missing when checkJavaScriptSection is called', () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  // @ts-expect-error: direct assignment for test
  instance.fileContentObject = {}
  expect(() => instance['checkJavaScriptSection']()).toThrow('JavaScript section is undefined')
})

it('biome I throws if json section is missing when checkJsonSection is called', () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  // @ts-expect-error: direct assignment for test
  instance.fileContentObject = {}
  expect(() => instance['checkJsonSection']()).toThrow('JSON section is undefined')
})

it('biome J throws if linter section is missing when checkLinterSection is called', () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  // @ts-expect-error: direct assignment for test
  instance.fileContentObject = {}
  expect(() => instance['checkLinterSection']()).toThrow('Linter section is undefined')
})

it('biome K throws if organizeImports section is missing when checkOrganizeImportsSection is called', () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  // @ts-expect-error: direct assignment for test
  instance.fileContentObject = {}
  expect(() => instance['checkOrganizeImportsSection']()).toThrow('organizeImports section is undefined')
})

it('biome L throws if vcs section is missing when checkVcsSection is called', () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  // @ts-expect-error: direct assignment for test
  instance.fileContentObject = {}
  expect(() => instance['checkVcsSection']()).toThrow('vcs section is undefined')
})

it('biome M throws if formatter section is missing when checkFormatterSection is called', () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  // @ts-expect-error: direct assignment for test
  instance.fileContentObject = {}
  expect(() => instance['checkFormatterSection']()).toThrow('Formatter section is undefined')
})

it('biome N fixes lineWidth if missing or too small and canFix is true', () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }), true)
  // @ts-expect-error: direct assignment for test
  instance.fileContentObject = { formatter: { indentStyle: 'space', indentWidth: 2, lineEnding: 'lf', lineWidth: 10 } }
  instance['checkFormatterSection']()
  // @ts-expect-error: test access
  expect(instance.fileContentObject?.formatter?.lineWidth).toBe(80)
})

it('biome O throws if overrides section is missing when checkOverridesSection is called', () => {
  const instance = new BiomeFile(repoCheckerPath, new ProjectData({ isQuiet: true }))
  // fileContentObject is undefined by default
  expect(() => instance['checkOverridesSection']()).toThrow('overrides section is undefined')
})
