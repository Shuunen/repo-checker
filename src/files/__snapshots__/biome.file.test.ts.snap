// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`biome A not using biome 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingBiome": false,
    "isUsingBun": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingKnip": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingPreact": false,
    "isUsingReact": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVitest": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "",
  "fileContentObject": undefined,
  "fileName": "",
  "originalFileContent": "",
  "passed": [],
  "warnings": [],
}
`;

exports[`biome B missing file 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingBiome": true,
    "isUsingBun": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingKnip": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingPreact": false,
    "isUsingReact": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVitest": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "",
  "fileName": "",
  "originalFileContent": "",
  "passed": [],
  "warnings": [],
}
`;

exports[`biome C empty file 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingBiome": true,
    "isUsingBun": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingKnip": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingPreact": false,
    "isUsingReact": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVitest": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "",
  "fileContentObject": {},
  "fileName": "",
  "originalFileContent": "",
  "passed": [],
  "warnings": [
    "use-the-correct-schema",
    "has-a-formatter-section",
    "has-a-javascript-section",
    "has-a-json-section",
    "has-a-linter-section",
    "has-a-organize-imports-section",
    "has-a-vcs-section",
    "has-an-overrides-section-array",
  ],
}
`;

exports[`biome D invalid json 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingBiome": true,
    "isUsingBun": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingKnip": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingPreact": false,
    "isUsingReact": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVitest": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "{ invalid json }",
  "fileName": "",
  "originalFileContent": "",
  "passed": [],
  "warnings": [],
}
`;

exports[`biome E minimal valid biome.json 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingBiome": true,
    "isUsingBun": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingKnip": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingPreact": false,
    "isUsingReact": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVitest": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "{"$schema":"./node_modules/@biomejs/biome/configuration_schema.json"}",
  "fileContentObject": {
    "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
  },
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "use-the-correct-schema",
  ],
  "warnings": [
    "has-a-formatter-section",
    "has-a-javascript-section",
    "has-a-json-section",
    "has-a-linter-section",
    "has-a-organize-imports-section",
    "has-a-vcs-section",
    "has-an-overrides-section-array",
  ],
}
`;

exports[`biome F fix missing sections 1`] = `
{
  "canFix": true,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingBiome": true,
    "isUsingBun": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingKnip": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingPreact": false,
    "isUsingReact": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVitest": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "{
  "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
  "formatter": {
    "indentStyle": "space",
    "indentWidth": 2,
    "lineEnding": "lf",
    "lineWidth": 80
  },
  "javascript": {
    "formatter": {
      "arrowParentheses": "asNeeded",
      "bracketSpacing": true,
      "quoteStyle": "single",
      "semicolons": "asNeeded"
    }
  },
  "json": {
    "parser": {
      "allowComments": false,
      "allowTrailingCommas": false
    }
  },
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true,
      "style": {
        "useBlockStatements": "off"
      }
    }
  },
  "organizeImports": {
    "enabled": true
  },
  "overrides": [
    {
      "include": [
        "*.test.ts"
      ],
      "linter": {
        "rules": {
          "style": {
            "noUnusedTemplateLiteral": "off"
          }
        }
      }
    }
  ],
  "vcs": {
    "clientKind": "git",
    "enabled": true,
    "useIgnoreFile": true
  }
}",
  "fileContentObject": {
    "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
    "formatter": {
      "indentStyle": "space",
      "indentWidth": 2,
      "lineEnding": "lf",
      "lineWidth": 80,
    },
    "javascript": {
      "formatter": {
        "arrowParentheses": "asNeeded",
        "bracketSpacing": true,
        "quoteStyle": "single",
        "semicolons": "asNeeded",
      },
    },
    "json": {
      "parser": {
        "allowComments": false,
        "allowTrailingCommas": false,
      },
    },
    "linter": {
      "enabled": true,
      "rules": {
        "recommended": true,
        "style": {
          "useBlockStatements": "off",
        },
      },
    },
    "organizeImports": {
      "enabled": true,
    },
    "overrides": [
      {
        "include": [
          "*.test.ts",
        ],
        "linter": {
          "rules": {
            "style": {
              "noUnusedTemplateLiteral": "off",
            },
          },
        },
      },
    ],
    "vcs": {
      "clientKind": "git",
      "enabled": true,
      "useIgnoreFile": true,
    },
  },
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "use-the-correct-schema",
    "has-a-formatter-section",
    "has-a-javascript-section",
    "has-a-json-section",
    "has-a-linter-section",
    "has-a-organize-imports-section",
    "has-a-vcs-section",
    "has-an-overrides-section-array",
  ],
  "warnings": [
    "could-uses-space-in-formatter-indent-style",
    "could-uses-2-in-formatter-indent-width",
    "could-uses-lf-in-formatter-line-ending",
    "could-uses-more-than-80-in-formatter-line-width",
    "could-use-as-needed-in-javascript-formatter-arrow-parentheses",
    "could-use-true-in-javascript-formatter-bracket-spacing",
    "could-use-single-in-javascript-formatter-quote-style",
    "could-use-as-needed-in-javascript-formatter-semicolons",
    "could-use-false-in-json-parser-allow-comments",
    "could-use-false-in-json-parser-allow-trailing-commas",
    "could-use-true-in-linter-enabled",
    "could-use-all-preferred-but-more-strict-or-recommended-in-linter-rules",
    "could-use-off-in-linter-rules-style-use-block-statements",
    "could-use-true-in-organize-imports-enabled",
    "could-use-true-in-vcs-enabled",
    "could-use-git-in-vcs-client-kind",
    "could-use-true-in-vcs-use-ignore-file",
    "has-a-test-override",
  ],
}
`;

exports[`biome G full valid biome.json 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isCli": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": true,
    "isUsingBiome": true,
    "isUsingBun": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingKnip": false,
    "isUsingNyc": false,
    "isUsingOxc": false,
    "isUsingPreact": false,
    "isUsingReact": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingV8": false,
    "isUsingVitest": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "{"$schema":"./node_modules/@biomejs/biome/configuration_schema.json","formatter":{"indentStyle":"space","indentWidth":2,"lineEnding":"lf","lineWidth":120},"javascript":{"formatter":{"arrowParentheses":"asNeeded","bracketSpacing":true,"quoteStyle":"single","semicolons":"asNeeded"}},"json":{"parser":{"allowComments":false,"allowTrailingCommas":false}},"linter":{"enabled":true,"rules":{"recommended":true,"style":{"useBlockStatements":"off"}}},"organizeImports":{"enabled":true},"vcs":{"clientKind":"git","enabled":true,"useIgnoreFile":true}}",
  "fileContentObject": {
    "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
    "formatter": {
      "indentStyle": "space",
      "indentWidth": 2,
      "lineEnding": "lf",
      "lineWidth": 120,
    },
    "javascript": {
      "formatter": {
        "arrowParentheses": "asNeeded",
        "bracketSpacing": true,
        "quoteStyle": "single",
        "semicolons": "asNeeded",
      },
    },
    "json": {
      "parser": {
        "allowComments": false,
        "allowTrailingCommas": false,
      },
    },
    "linter": {
      "enabled": true,
      "rules": {
        "recommended": true,
        "style": {
          "useBlockStatements": "off",
        },
      },
    },
    "organizeImports": {
      "enabled": true,
    },
    "vcs": {
      "clientKind": "git",
      "enabled": true,
      "useIgnoreFile": true,
    },
  },
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "use-the-correct-schema",
    "has-a-formatter-section",
    "could-uses-space-in-formatter-indent-style",
    "could-uses-2-in-formatter-indent-width",
    "could-uses-lf-in-formatter-line-ending",
    "could-uses-more-than-80-in-formatter-line-width",
    "has-a-javascript-section",
    "could-use-as-needed-in-javascript-formatter-arrow-parentheses",
    "could-use-true-in-javascript-formatter-bracket-spacing",
    "could-use-single-in-javascript-formatter-quote-style",
    "could-use-as-needed-in-javascript-formatter-semicolons",
    "has-a-json-section",
    "could-use-false-in-json-parser-allow-comments",
    "could-use-false-in-json-parser-allow-trailing-commas",
    "has-a-linter-section",
    "could-use-true-in-linter-enabled",
    "could-use-all-preferred-but-more-strict-or-recommended-in-linter-rules",
    "could-use-off-in-linter-rules-style-use-block-statements",
    "has-a-organize-imports-section",
    "could-use-true-in-organize-imports-enabled",
    "has-a-vcs-section",
    "could-use-true-in-vcs-enabled",
    "could-use-git-in-vcs-client-kind",
    "could-use-true-in-vcs-use-ignore-file",
  ],
  "warnings": [
    "has-an-overrides-section-array",
  ],
}
`;
